src/expr/impl/src/scalar/cast.rs

```diff
@@ -40,32 +40,37 @@ use thiserror_ext::AsReport;
 pub fn str_parse<T>(elem: &str, ctx: &Context) -> Result<T>
 where
     T: FromStr,
-    <T as FromStr>::Err: std::fmt::Display,
 {
-    elem.trim().parse().map_err(|err: <T as FromStr>::Err| {
-        ExprError::Parse(format!("{} {}", ctx.return_type, err).into())
+    elem.trim().parse().ok().with_context(|| {
```

Comment:
`.ok()` is not necessary here? 🤔

Reply:
If removed, a complex type constraint would be required.
```
error[E0277]: `<T as FromStr>::Err` cannot be sent between threads safely
  --> src/expr/impl/src/scalar/cast.rs:44:25
   |
44 |     elem.trim().parse().with_context(|| {
   |                         ^^^^^^^^^^^^ `<T as FromStr>::Err` cannot be sent between threads safely
   |
   = help: the trait `std::marker::Send` is not implemented for `<T as FromStr>::Err`
   = note: required for `std::result::Result<T, <T as FromStr>::Err>` to implement `anyhow::Context<T, <T as FromStr>::Err>`
help: consider further restricting the associated type
   |
42 |     T: FromStr, <T as FromStr>::Err: std::marker::Send
   |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```